modules/react/collection/index.ts

```diff
@@ -10,7 +10,8 @@ export * from './lib/useListItemRovingFocus';
 export * from './lib/useListItemSelect';
 export * from './lib/useListModel';
 export * from './lib/useGridModel';
-export {ListBox} from './lib/ListBox';
+export * from './lib/useListItemAllowChildStrings';
+export {ListBox, ListBoxProps} from './lib/ListBox';
```

Comment:
I didn't see where we needed `ListBoxProps` to be added as an export. Also thought we could use `ExtractProps` on the consumer side if necessary.

Reply:
I did not mean to add this change to this PR. I think it is used in the Combobox changes. I think the reason why I did this was because `ExtractProps<typeof ListBox, never>` returns the model as part of the prop interface. `ExtractProps` was made before model components existed and it isn't clear what it should return with model components.

I don't know what will be most common - wanting the `model` on the prop interface or not wanting it. In the case of a combobox component, I wanted to use the `children` and other props of the `ListBox` without defining it again, but I need a new model. That would look like this:

```ts
export ComboboxListProps extends Omit<ExtractProps<typeof ListBox, never>, 'model'> {
```

Most other components that have the same interface as `ListBox` manually type the `children`:
- https://github.com/Workday/canvas-kit/blob/99b29390ab73aa15117d6fc147c23d814ffc43d3/modules/react/tabs/lib/TabsList.tsx#L34
- https://github.com/Workday/canvas-kit/blob/99b29390ab73aa15117d6fc147c23d814ffc43d3/modules/react/breadcrumbs/lib/BreadcrumbsList.tsx#L22
- https://github.com/Workday/canvas-kit/blob/99b29390ab73aa15117d6fc147c23d814ffc43d3/modules/preview-react/segmented-control/lib/SegmentedControlList.tsx#L15

I was experimenting with a different paradigm where extending a `ListBox` component would involve extending the props instead. Thoughts?